include/glaze/containers/inplace_vector.hpp

```diff
@@ -0,0 +1,1111 @@
+         }
+
+         // Fast path for contiguous ranges of trivially copyable types
+         if constexpr (std::is_trivially_copyable_v<T> && std::ranges::contiguous_range<R> &&
+                       std::same_as<std::remove_cvref_t<std::ranges::range_reference_t<R>>, T>) {
+            auto count = std::ranges::size(rg);
+            if (count > 0) {
+               std::memcpy(this->storage, std::ranges::data(rg), count * sizeof(T));
+               this->set_storage_size(static_cast<size_type>(count));
+            }
+         }
+         else {
+            // General path
+            for (auto&& item : rg) {
+               if (this->storage_size() >= N) GLZ_THROW_OR_ABORT(std::bad_alloc());
+
+               std::construct_at(this->data_ptr() + this->storage_size(), std::forward<decltype(item)>(item));
+               this->set_storage_size(this->storage_size() + 1);
+            }
+         }
+      }
+
+      constexpr void assign(size_type n, const T& value) // freestanding-deleted
+      {
+         if (n > N) GLZ_THROW_OR_ABORT(std::bad_alloc());
+
+         this->clear();
+
+         if constexpr (std::is_trivially_copyable_v<T>) {
+            if (n > 0) {
+               // Construct first element, then use memcpy for the rest
+               std::construct_at(this->data_ptr(), value);
+               for (size_type i = 1; i < n; ++i) {
+                  std::memcpy(this->data_ptr() + i, this->data_ptr(), sizeof(T));
+               }
+               this->set_storage_size(n);
+            }
+         }
+         else {
+            for (size_type i = 0; i < n; ++i) {
+               std::construct_at(this->data_ptr() + this->storage_size(), value);
+               this->set_storage_size(this->storage_size() + 1);
+            }
+         }
+      }
+
+      // TODO add assign(initializer_list<T>)
+
+      // Capacity
+
+      constexpr void resize(size_type sz) // freestanding-deleted
+      {
+         if (sz > N) GLZ_THROW_OR_ABORT(std::bad_alloc());
+
+         if (sz > this->storage_size()) {
+            // Value-initialize the new elements
+            std::uninitialized_value_construct_n(this->data_ptr() + this->storage_size(), sz - this->storage_size());
+            this->set_storage_size(sz);
+         }
+         else if (sz < this->storage_size()) {
+            // Destroy excess elements
+            this->destroy_range(sz, this->storage_size());
+            this->set_storage_size(sz);
+         }
+      }
+
+      constexpr void resize(size_type sz, const T& c) // freestanding-deleted
+      {
+         if (sz > N) GLZ_THROW_OR_ABORT(std::bad_alloc());
+
+         if (sz > this->storage_size()) {
+            if constexpr (std::is_trivially_copyable_v<T>) {
+               // For trivially copyable types, construct one and copy the rest
+               if (this->storage_size() < sz) {
+                  if (this->storage_size() == 0 && sz > 0) {
+                     std::construct_at(this->data_ptr(), c);
```

Comment:
I didn't get it. At 652 it is OK to memcpy the &c, then here why constructed the first element and then memcpy that (isn't it easier to memcpy &C here too)?